pkg/utils/packet.go

```diff
@@ -0,0 +1,176 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"syscall"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+)
+
+var (
+	arpPacketName    = "ARP"
+	icmpV6PacketName = "ICMPv6"
+)
+
+// htons converts an uint16 from host to network byte order.
+func htons(i uint16) uint16 {
+	return (i<<8)&0xff00 | i>>8
+}
+
+// formatPacketFieldWriteError builds an error string for the cases when writing to a field of a packet fails.
+func formatPacketFieldWriteError(field string, packetType string, writeErr error) error {
+	return fmt.Errorf("failed to write the %s field in the %s packet: %v", field, packetType, writeErr)
+}
+
+// IsValidMACAddress checks if net.HardwareAddr is a valid MAC address.
+func IsValidMACAddress(addr net.HardwareAddr) bool {
+	return (len(addr) == 6)
+}
+
+// IsIPv4 checks if a net.IP is an IPv4 address.
+func IsIPv4(ip net.IP) bool {
+	return (ip.To4() != nil)
+}
+
+// IsIPv6 checks if a net.IP is an IPv6 address.
+func IsIPv6(ip net.IP) bool {
+	return ((ip.To4() == nil) && (ip.To16() != nil))
```

Comment:
Why return (...)? why not return ... 

Same comment for others.

Reply:
Just a personal preference for readability. I can make this minor change.

Reply:
Made the change.